syntax = "proto3";
// The purpose of this proto buffer file is to define the standard of communication between Master and Worker
// Feedback regarding field adjustment / rpc adjustments via @ryantangit or @birtext in class discord.

service SegmentEncodeService {
	rpc EncodeVideoSegment(stream EncodeSegmentRequest) returns (EncodeSegmentResponse) {}
	rpc RetrieveVideoSegment(RetrieveSegmentRequest) returns (stream RetrieveSegmentResponse) {}
}

message EncodeSegmentRequest {
	bytes video_data = 1;
	string segment_name = 2;
	string video_name = 3; //Serves as the directory in node 
}

message EncodeSegmentResponse {
	bool success = 1;
	string status_message = 2;
	string segment_name = 3;
}

message RetrieveSegmentRequest {
	string segment_name = 1;
	string video_name = 2;
}

message RetrieveSegmentResponse {
	bytes video_data = 1;	
}
